File: src/main/java/com/gzst/shopping/service/BrandService.java
package com.gzst.shopping.service;

import com.gzst.shopping.model.Brand;
import com.gzst.shopping.utils.PageResult;
import com.gzst.shopping.utils.Result;

import java.util.List;

public interface BrandService {
    public List<Brand> findAllBrands();

    public PageResult<Brand> findPage(Integer pageNo, Integer pageSize);

    public Result saveBrand(Brand brand);

    public Brand findOne(Integer id);

    public Result updateBrand(Brand brand);

    public Result deleteBrand(Integer[] id);
}

File: src/main/resources/static/js/service/brandService.js
app.service('brandService', function ($http) {
    //service负责与后台进数据交互
    this.findAll = function () {
        return $http.get('../brand/getAllBrands');
    }
    this.update = function (entity) {
        return $http.post('../brand/updateBrand', entity);
    }
    this.addBrand = function (entity) {
        return $http.post('../brand/addBrand', entity);
    }
    this.findPage = function (pageNo, pageSize) {
        return $http.get('../brand/findPage?pageNo=' + pageNo + '&pageSize=' + pageSize);
    }
    this.findOne = function (id) {
        return $http.get('../brand/findOne?id=' + id);
    }
})